src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -2,36 +2,107 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 import re
+from dataclasses import dataclass
 from typing import Match, Optional, Tuple, cast
 
+from pants.backend.python.dependency_inference.module_mapper import PythonModule, PythonModuleOwners
+from pants.backend.python.dependency_inference.rules import import_rules
 from pants.backend.python.target_types import InterpreterConstraintsField, PythonSources
 from pants.core.goals.package import OutputPathField
 from pants.engine.addresses import Address
+from pants.engine.rules import Get, collect_rules, rule
 from pants.engine.target import (
     COMMON_TARGET_FIELDS,
+    AsyncFieldMixin,
     Dependencies,
+    InjectDependenciesRequest,
+    InjectedDependencies,
     InvalidFieldException,
     StringField,
     Target,
+    WrappedTarget,
 )
+from pants.engine.unions import UnionRule
+from pants.option.subsystem import Subsystem
 
 
-class PythonAwsLambdaSources(PythonSources):
-    expected_num_files = range(0, 1)
+class PythonAwsLambdaDefaults(Subsystem):
+    """Default settings for the `python_awslambda` target."""
 
+    options_scope = "python-awslambda-defaults"
 
-class PythonAwsLambdaDependencies(Dependencies):
-    supports_transitive_excludes = True
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--infer-dependencies",
+            advanced=True,
+            type=bool,
+            default=True,
+            help="Infer a dependency on the module specified by the `handler` field.",
+        )
+
+    @property
+    def infer_dependencies(self) -> bool:
+        return cast(bool, self.options.infer_dependencies)
+
+
+class PythonAwsLambdaSources(PythonSources):
+    expected_num_files = range(0, 2)
```

Comment:
Maybe this is for a followup PR (as it requires a little bit of a change in `target.py`) but this should just be `(0, 1)`. Being too-clever-by-half might be what led to the bug in the first place. 

Also, `range` constructs a generator-like object, and is a rather heavyweight way of just providing what will always be a teeny tiny tuple in practice.

Reply:
We are plugin authors and we somehow got this wrong...

Reply:
Even if we don't fix this, let's take it as a lesson to KISS...

Reply:
In the future, that is.